File: Bit Manipulation/2419.Longest Subarray With Maximum Bitwise AND/Solution.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        mx = cur = ans = 0
        for i, v in enumerate(nums):
            if v > mx:
                cur = ans = 1
                mx = v
            elif v == mx:
                cur += 1
            else:
                cur = 0
            ans = max(ans, cur)
        return ans

File: Bit Manipulation/2419.Longest Subarray With Maximum Bitwise AND/Solution.java
class Solution {
    public int longestSubarray(int[] nums) {
        int mx = 0, cur = 0, ans = 0;
        for (int v : nums) {
            if (v > mx) {
                cur = 1;
                ans = 1;
                mx = v;
            } else if (v == mx) {
                cur++;
            } else {
                cur = 0;
            }
            ans = Math.max(ans, cur);
        }
        return ans;
    }
}
